arrivals/src/main/scala/com/pagerduty/arrivals/impl/filter/FilterDirectives.scala

```diff
@@ -0,0 +1,46 @@
+package com.pagerduty.arrivals.impl.filter
+
+import akka.http.scaladsl.server.{Directive, Directive0, Directive1, RouteResult}
+import com.pagerduty.arrivals.api.filter._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.RouteResult.Complete
+
+import scala.concurrent.Future
+
+object FilterDirectives {
+
+  def filterRequest[RequestData](requestFilter: RequestFilter[RequestData], requestData: RequestData): Directive0 = {
+    extractRequestFilterResult(requestFilter, requestData).flatMap {
+      case Right(filteredRequest) => mapRequest(_ => filteredRequest)
+      case Left(response)         => complete(response)
```

Comment:
would be nicer with a sum type here for the FilterResult

Reply:
Yeah, let's explore this in a different PR.